[2024-11-08 12:57:05,881: INFO: main: Welcome to custom log]
[2024-11-08 12:59:12,690: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-08 12:59:12,694: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-08 17:24:04,349: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-08 17:24:04,358: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-08 17:32:19,390: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-08 17:32:19,396: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-08 17:32:19,400: INFO: common: created directory at: artifacts]
[2024-11-08 17:32:19,403: INFO: common: created directory at: artifacts/data_ingestion]
[2024-11-08 17:32:58,396: INFO: 2388877383: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 22756660
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "07b1147bb12d018cef6446014bc4d1b098f8a8db79c8f5e8866fed33abc1a055"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 7C62:1B13A2:A3502:C6553:672DFDCF
Accept-Ranges: bytes
Date: Fri, 08 Nov 2024 12:02:26 GMT
Via: 1.1 varnish
X-Served-By: cache-maa10228-MAA
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1731067345.401845,VS0,VE1548
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 16b196a200316f27618dd370567ccacf1e94b4c8
Expires: Fri, 08 Nov 2024 12:07:26 GMT
Source-Age: 1

]
[2024-11-08 17:38:37,628: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-08 17:38:37,628: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-08 17:38:37,628: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-08 17:38:37,641: INFO: common: created directory at: artifacts]
[2024-11-08 17:38:37,641: INFO: common: created directory at: artifacts/data_ingestion]
[2024-11-08 17:38:37,643: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-08 17:38:37,938: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 12:21:19,022: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-09 12:21:19,030: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-09 12:21:19,031: INFO: common: created directory at: artifacts]
[2024-11-09 12:21:19,036: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 12:23:01,660: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-09 12:23:01,667: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-09 12:23:01,671: INFO: common: created directory at: artifacts]
[2024-11-09 12:23:01,673: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 13:02:08,727: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-09 13:02:08,732: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-09 13:02:08,733: INFO: common: created directory at: artifacts]
[2024-11-09 13:02:08,735: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 13:03:07,321: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-09 13:03:07,325: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-09 13:03:07,327: INFO: common: created directory at: artifacts]
[2024-11-09 13:03:07,329: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 13:28:10,596: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 13:28:10,606: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 13:28:10,610: INFO: common: created directory at: artifacts]
[2024-11-09 13:29:47,020: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 13:29:47,041: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 13:29:47,045: INFO: common: created directory at: artifacts]
[2024-11-09 13:29:47,047: INFO: common: created directory at: artifacts\prepare_base_model]
[2024-11-09 13:31:24,244: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 13:31:24,254: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 13:31:24,258: INFO: common: created directory at: artifacts]
[2024-11-09 13:31:24,260: INFO: common: created directory at: artifacts\prepare_base_model]
[2024-11-09 13:31:53,180: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 13:31:53,193: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 13:31:53,196: INFO: common: created directory at: artifacts]
[2024-11-09 13:31:53,200: INFO: common: created directory at: artifacts\prepare_base_model]
[2024-11-09 13:32:53,385: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 13:32:53,396: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 13:32:53,401: INFO: common: created directory at: artifacts]
[2024-11-09 13:32:53,402: INFO: common: created directory at: artifacts\prepare_base_model]
[2024-11-09 13:33:56,774: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 13:33:56,782: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 13:33:56,785: INFO: common: created directory at: artifacts]
[2024-11-09 13:33:56,788: INFO: common: created directory at: artifacts\prepare_base_model]
[2024-11-09 13:34:02,775: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 13:34:02,785: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 13:34:02,789: INFO: common: created directory at: artifacts]
[2024-11-09 13:34:02,791: INFO: common: created directory at: artifacts\prepare_base_model]
[2024-11-09 13:35:30,090: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 13:35:30,097: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 13:35:30,101: INFO: common: created directory at: artifacts]
[2024-11-09 13:36:21,695: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 13:36:21,704: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 13:36:21,708: INFO: common: created directory at: artifacts]
[2024-11-09 13:36:37,442: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 13:36:37,449: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 13:36:37,455: INFO: common: created directory at: artifacts]
[2024-11-09 15:14:59,299: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:14:59,308: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:14:59,308: INFO: common: created directory at: artifacts]
[2024-11-09 15:16:47,543: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:16:47,556: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:16:47,559: INFO: common: created directory at: artifacts]
[2024-11-09 15:17:11,781: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:17:11,785: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:17:11,790: INFO: common: created directory at: artifacts]
[2024-11-09 15:18:27,204: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:18:27,213: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:18:27,213: INFO: common: created directory at: artifacts]
[2024-11-09 15:19:39,546: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:19:39,555: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:19:39,555: INFO: common: created directory at: artifacts]
[2024-11-09 15:19:39,563: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 15:20:41,562: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:20:41,572: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:20:41,577: INFO: common: created directory at: artifacts]
[2024-11-09 15:20:41,577: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 15:21:01,636: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:21:01,642: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:21:01,648: INFO: common: created directory at: artifacts]
[2024-11-09 15:21:01,653: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 15:21:41,528: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:21:41,541: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:21:41,545: INFO: common: created directory at: artifacts]
[2024-11-09 15:21:41,547: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 15:21:48,733: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:21:48,743: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:21:48,746: INFO: common: created directory at: artifacts]
[2024-11-09 15:21:48,748: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 15:39:09,468: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 15:39:09,468: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 15:39:20,399: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 15:39:20,399: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-09 15:39:20,411: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-09 15:39:20,411: INFO: common: created directory at: artifacts]
[2024-11-09 15:39:20,414: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 15:39:59,548: INFO: data_ingestion: artifacts\data_ingestion\data.zip download! with following info: 
Connection: close
Content-Length: 22756660
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "07b1147bb12d018cef6446014bc4d1b098f8a8db79c8f5e8866fed33abc1a055"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 63F4:20182A:91334:C38A1:672F34D5
Accept-Ranges: bytes
Date: Sat, 09 Nov 2024 10:09:27 GMT
Via: 1.1 varnish
X-Served-By: cache-bom4740-BOM
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1731146966.118528,VS0,VE1481
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: f80ace7716ebf50732c9bfaba4ec6303f5874891
Expires: Sat, 09 Nov 2024 10:14:27 GMT
Source-Age: 1

]
[2024-11-09 15:39:59,869: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 15:39:59,869: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 15:39:59,876: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-09 15:39:59,880: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-09 15:39:59,880: INFO: common: created directory at: artifacts]
[2024-11-09 15:39:59,884: INFO: common: created directory at: artifacts\prepare_base_model]
[2024-11-09 15:39:59,884: ERROR: main: "'IMAGE_SIZE'"]
Traceback (most recent call last):
  File "box\\box.py", line 593, in box.box.Box.__getitem__
KeyError: 'IMAGE_SIZE'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 25, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_02_prepare_base_model.py", line 13, in main
    prepare_base_model_config = config_manager.get_prepare_base_model_config()
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Projects\Steganalysis\src\Steganalysis\config\configuration.py", line 46, in get_prepare_base_model_config
    params_image_size=tuple(self.params["IMAGE_SIZE"]),   # Convert to tuple if needed
                            ~~~~~~~~~~~^^^^^^^^^^^^^^
  File "box\\box.py", line 620, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'IMAGE_SIZE'"
[2024-11-09 15:42:26,837: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 15:42:26,837: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 15:42:29,043: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 15:42:29,047: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-09 15:42:29,065: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-09 15:42:29,072: INFO: common: created directory at: artifacts]
[2024-11-09 15:42:29,072: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 15:42:29,080: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 15:42:29,373: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 15:42:29,373: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 15:42:29,378: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-09 15:42:29,384: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-09 15:42:29,384: INFO: common: created directory at: artifacts]
[2024-11-09 15:42:29,389: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 15:42:29,391: ERROR: main: PrepareBaseModelConfig.__init__() got an unexpected keyword argument 'model_type']
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 25, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_02_prepare_base_model.py", line 13, in main
    prepare_base_model_config = config_manager.get_prepare_base_model_config()
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Projects\Steganalysis\src\Steganalysis\config\configuration.py", line 41, in get_prepare_base_model_config
    prepare_base_model_config = PrepareBaseModelConfig(
                                ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: PrepareBaseModelConfig.__init__() got an unexpected keyword argument 'model_type'
[2024-11-09 15:44:01,603: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 15:44:01,611: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 15:44:03,807: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 15:44:03,817: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-09 15:44:03,822: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-09 15:44:03,824: INFO: common: created directory at: artifacts]
[2024-11-09 15:44:03,824: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 15:44:03,824: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 15:44:04,096: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 15:44:04,096: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 15:44:04,103: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-09 15:44:04,108: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-09 15:44:04,111: INFO: common: created directory at: artifacts]
[2024-11-09 15:44:04,114: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 15:44:04,115: ERROR: main: PrepareBaseModelConfig.__init__() got an unexpected keyword argument 'optimizer']
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 25, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_02_prepare_base_model.py", line 13, in main
    prepare_base_model_config = config_manager.get_prepare_base_model_config()
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Projects\Steganalysis\src\Steganalysis\config\configuration.py", line 41, in get_prepare_base_model_config
    prepare_base_model_config = PrepareBaseModelConfig(
                                ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: PrepareBaseModelConfig.__init__() got an unexpected keyword argument 'optimizer'
[2024-11-09 15:45:24,259: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 15:45:24,260: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 15:45:25,249: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 15:45:25,255: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-09 15:45:25,258: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-09 15:45:25,258: INFO: common: created directory at: artifacts]
[2024-11-09 15:45:25,259: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 15:45:25,259: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 15:45:25,378: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 15:45:25,378: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 15:45:25,379: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-09 15:45:25,382: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-09 15:45:25,382: INFO: common: created directory at: artifacts]
[2024-11-09 15:45:25,382: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 15:45:25,384: ERROR: main: PrepareBaseModelConfig.__init__() got an unexpected keyword argument 'batch_size']
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 25, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_02_prepare_base_model.py", line 13, in main
    prepare_base_model_config = config_manager.get_prepare_base_model_config()
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Projects\Steganalysis\src\Steganalysis\config\configuration.py", line 41, in get_prepare_base_model_config
    prepare_base_model_config = PrepareBaseModelConfig(
                                ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: PrepareBaseModelConfig.__init__() got an unexpected keyword argument 'batch_size'
[2024-11-09 15:46:39,380: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 15:46:39,380: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 15:46:40,421: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 15:46:40,424: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-09 15:46:40,426: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-09 15:46:40,426: INFO: common: created directory at: artifacts]
[2024-11-09 15:46:40,426: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 15:46:40,427: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 15:46:40,548: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 15:46:40,550: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 15:46:40,552: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-11-09 15:46:40,554: INFO: common: yaml file: params.yaml loaded successfully]
[2024-11-09 15:46:40,554: INFO: common: created directory at: artifacts]
[2024-11-09 15:46:40,555: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 15:46:40,555: ERROR: main: PrepareBaseModelConfig.__init__() missing 1 required positional argument: 'params_learning_rate']
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 25, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_02_prepare_base_model.py", line 13, in main
    prepare_base_model_config = config_manager.get_prepare_base_model_config()
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Projects\Steganalysis\src\Steganalysis\config\configuration.py", line 41, in get_prepare_base_model_config
    prepare_base_model_config = PrepareBaseModelConfig(
                                ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: PrepareBaseModelConfig.__init__() missing 1 required positional argument: 'params_learning_rate'
[2024-11-09 15:47:51,675: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 15:47:51,675: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 15:47:52,666: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 15:47:52,671: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:47:52,674: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:47:52,675: INFO: common: created directory at: artifacts]
[2024-11-09 15:47:52,675: ERROR: main: 'ConfigurationManager' object has no attribute 'get_data_ingestion_config']
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 14, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_01_data_ingestion.py", line 14, in main
    data_ingestion_config = config.get_data_ingestion_config()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ConfigurationManager' object has no attribute 'get_data_ingestion_config'
[2024-11-09 15:50:56,256: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 15:50:56,256: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 15:50:58,571: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 15:50:58,585: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:50:58,591: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:50:58,597: INFO: common: created directory at: artifacts]
[2024-11-09 15:50:58,597: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 15:50:58,599: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 15:50:58,883: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 15:50:58,883: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 15:50:58,890: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:50:58,898: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:50:58,900: INFO: common: created directory at: artifacts]
[2024-11-09 15:50:58,900: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 15:50:58,902: ERROR: main: PrepareBaseModelConfig.__init__() got an unexpected keyword argument 'model_type']
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 25, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_02_prepare_base_model.py", line 13, in main
    prepare_base_model_config = config_manager.get_prepare_base_model_config()
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Projects\Steganalysis\src\Steganalysis\config\configuration.py", line 44, in get_prepare_base_model_config
    prepare_base_model_config = PrepareBaseModelConfig(
                                ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: PrepareBaseModelConfig.__init__() got an unexpected keyword argument 'model_type'
[2024-11-09 15:55:05,573: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 15:55:05,573: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 15:55:08,005: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 15:55:08,009: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:55:08,016: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:55:08,016: INFO: common: created directory at: artifacts]
[2024-11-09 15:55:08,016: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 15:55:08,019: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 15:55:08,325: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 15:55:08,326: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 15:55:08,328: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:55:08,336: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:55:08,339: INFO: common: created directory at: artifacts]
[2024-11-09 15:55:08,340: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 15:55:12,285: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 15:57:50,108: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 15:57:50,116: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 15:57:50,123: INFO: common: created directory at: artifacts]
[2024-11-09 15:57:50,123: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 16:02:44,529: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 16:02:44,536: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 16:02:47,242: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 16:02:47,249: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 16:02:47,255: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 16:02:47,258: INFO: common: created directory at: artifacts]
[2024-11-09 16:02:47,260: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 16:02:47,260: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 16:02:47,565: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 16:02:47,565: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 16:02:47,567: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 16:02:47,573: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 16:02:47,578: INFO: common: created directory at: artifacts]
[2024-11-09 16:02:47,580: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 16:02:53,810: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 16:03:29,910: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 16:03:29,910: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 16:03:32,212: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 16:03:32,221: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 16:03:32,230: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 16:03:32,232: INFO: common: created directory at: artifacts]
[2024-11-09 16:03:32,233: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 16:12:12,162: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 16:12:12,162: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 16:12:14,560: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 16:12:14,568: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 16:12:14,571: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 16:12:14,577: INFO: common: created directory at: artifacts]
[2024-11-09 16:12:14,579: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 16:12:14,579: INFO: data_ingestion: File already exists of size: 12.66 MB]
[2024-11-09 16:12:14,579: ERROR: main: File is not a zip file]
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 14, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_01_data_ingestion.py", line 17, in main
    data_ingestion.extract_zip_file()
  File "E:\Projects\Steganalysis\src\Steganalysis\components\data_ingestion.py", line 77, in extract_zip_file
    with zipfile.ZipFile(self.config.local_data_file, 'r') as zip_ref:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\System_Files\Anaconda\Lib\zipfile\__init__.py", line 1349, in __init__
    self._RealGetContents()
  File "E:\System_Files\Anaconda\Lib\zipfile\__init__.py", line 1416, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
[2024-11-09 16:12:48,863: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 16:12:48,863: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 16:12:51,937: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 16:12:51,942: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 16:12:51,948: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 16:12:51,948: INFO: common: created directory at: artifacts]
[2024-11-09 16:12:51,948: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 16:24:00,069: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 16:24:00,069: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 16:24:01,070: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 16:24:01,071: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 16:24:01,071: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 16:24:01,071: INFO: common: created directory at: artifacts]
[2024-11-09 16:24:01,071: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 16:24:37,004: INFO: data_ingestion: artifacts\data_ingestion\data.zip download! with following info: 
Connection: close
Content-Length: 22756660
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "07b1147bb12d018cef6446014bc4d1b098f8a8db79c8f5e8866fed33abc1a055"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 63F4:20182A:91334:C38A1:672F34D5
Accept-Ranges: bytes
Date: Sat, 09 Nov 2024 10:54:05 GMT
Via: 1.1 varnish
X-Served-By: cache-bom4722-BOM
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1731149646.577533,VS0,VE336
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: f9d0e494f92a673553de79e995a6de1754c0062d
Expires: Sat, 09 Nov 2024 10:59:05 GMT
Source-Age: 0

]
[2024-11-09 16:24:37,143: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 16:24:37,143: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 16:24:37,145: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 16:24:37,146: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 16:24:37,146: INFO: common: created directory at: artifacts]
[2024-11-09 16:24:37,149: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 16:24:39,895: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 16:56:44,307: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 16:56:44,310: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 16:56:44,313: INFO: common: created directory at: artifacts]
[2024-11-09 17:00:08,252: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:00:08,256: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:00:08,257: INFO: common: created directory at: artifacts]
[2024-11-09 17:18:57,657: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 17:18:57,657: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 17:19:00,260: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 17:19:00,265: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:19:00,271: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:19:00,273: INFO: common: created directory at: artifacts]
[2024-11-09 17:19:00,273: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 17:19:00,273: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 17:19:00,598: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 17:19:00,598: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 17:19:00,602: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:19:00,608: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:19:00,612: INFO: common: created directory at: artifacts]
[2024-11-09 17:19:00,613: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 17:19:07,219: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 17:19:07,219: INFO: main: >>>>>> stage Prepare Callbacks started <<<<<<]
[2024-11-09 17:19:07,222: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:19:07,228: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:19:07,233: INFO: common: created directory at: artifacts]
[2024-11-09 17:19:07,233: INFO: common: created directory at: artifacts/prepare_callbacks]
[2024-11-09 17:19:07,235: INFO: main: >>>>>> stage Prepare Callbacks completed <<<<<<

x==========x]
[2024-11-09 17:38:52,263: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 17:38:52,264: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 17:38:54,541: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 17:38:54,546: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:38:54,549: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:38:54,552: INFO: common: created directory at: artifacts]
[2024-11-09 17:38:54,553: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 17:38:54,555: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 17:38:54,824: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 17:38:54,825: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 17:38:54,827: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:38:54,834: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:38:54,834: INFO: common: created directory at: artifacts]
[2024-11-09 17:38:54,834: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 17:39:00,011: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 17:39:00,014: INFO: main: >>>>>> stage Prepare Callbacks started <<<<<<]
[2024-11-09 17:39:00,018: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:39:00,030: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:39:00,032: INFO: common: created directory at: artifacts]
[2024-11-09 17:39:00,032: INFO: common: created directory at: artifacts/prepare_callbacks]
[2024-11-09 17:39:00,034: INFO: main: >>>>>> stage Prepare Callbacks completed <<<<<<

x==========x]
[2024-11-09 17:39:00,035: INFO: main: >>>>>> stage Model Training started <<<<<<]
[2024-11-09 17:39:00,045: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:39:00,052: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:39:00,053: INFO: common: created directory at: artifacts]
[2024-11-09 17:39:00,053: INFO: common: created directory at: artifacts/training]
[2024-11-09 17:39:00,056: ERROR: main: File not found: filepath=artifacts\prepare_base_model\base_model_updated.keras. Please ensure the file is an accessible `.keras` zip file.]
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 49, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_04_model_training.py", line 19, in main
    model = load_model(model_training_config.model_path)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\System_Files\Anaconda\Lib\site-packages\keras\src\saving\saving_api.py", line 200, in load_model
    raise ValueError(
ValueError: File not found: filepath=artifacts\prepare_base_model\base_model_updated.keras. Please ensure the file is an accessible `.keras` zip file.
[2024-11-09 17:46:01,232: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 17:46:01,234: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 17:46:03,620: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 17:46:03,623: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:46:03,628: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:46:03,632: INFO: common: created directory at: artifacts]
[2024-11-09 17:46:03,632: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 17:46:03,634: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 17:46:03,913: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 17:46:03,913: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 17:46:03,918: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:46:03,923: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:46:03,925: INFO: common: created directory at: artifacts]
[2024-11-09 17:46:03,925: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 17:46:09,609: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 17:46:09,611: INFO: main: >>>>>> stage Prepare Callbacks started <<<<<<]
[2024-11-09 17:46:09,614: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:46:09,619: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:46:09,622: INFO: common: created directory at: artifacts]
[2024-11-09 17:46:09,622: INFO: common: created directory at: artifacts/prepare_callbacks]
[2024-11-09 17:46:09,624: INFO: main: >>>>>> stage Prepare Callbacks completed <<<<<<

x==========x]
[2024-11-09 17:46:09,624: INFO: main: >>>>>> stage Model Training started <<<<<<]
[2024-11-09 17:46:09,628: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:46:09,633: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:46:09,633: INFO: common: created directory at: artifacts]
[2024-11-09 17:46:09,633: INFO: common: created directory at: artifacts/training]
[2024-11-09 17:46:14,570: ERROR: main: 'ModelTrainingConfig' object has no attribute 'image_size']
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 49, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_04_model_training.py", line 29, in main
    target_size=model_training_config.image_size[:2],
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ModelTrainingConfig' object has no attribute 'image_size'
[2024-11-09 17:49:31,105: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 17:49:31,105: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 17:49:33,442: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 17:49:33,454: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:49:33,462: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:49:33,463: INFO: common: created directory at: artifacts]
[2024-11-09 17:49:33,463: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 17:49:33,463: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 17:49:33,744: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 17:49:33,744: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 17:49:33,751: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:49:33,757: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:49:33,757: INFO: common: created directory at: artifacts]
[2024-11-09 17:49:33,757: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 17:49:39,279: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 17:49:39,285: INFO: main: >>>>>> stage Prepare Callbacks started <<<<<<]
[2024-11-09 17:49:39,291: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:49:39,300: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:49:39,302: INFO: common: created directory at: artifacts]
[2024-11-09 17:49:39,303: INFO: common: created directory at: artifacts/prepare_callbacks]
[2024-11-09 17:49:39,303: INFO: main: >>>>>> stage Prepare Callbacks completed <<<<<<

x==========x]
[2024-11-09 17:49:39,303: INFO: main: >>>>>> stage Model Training started <<<<<<]
[2024-11-09 17:49:39,311: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:49:39,314: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:49:39,319: INFO: common: created directory at: artifacts]
[2024-11-09 17:49:39,319: ERROR: main: ModelTrainingConfig.__init__() got an unexpected keyword argument 'image_size']
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 49, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_04_model_training.py", line 16, in main
    model_training_config = config_manager.get_model_training_config()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Projects\Steganalysis\src\Steganalysis\config\configuration.py", line 74, in get_model_training_config
    model_training_config = ModelTrainingConfig(
                            ^^^^^^^^^^^^^^^^^^^^
TypeError: ModelTrainingConfig.__init__() got an unexpected keyword argument 'image_size'
[2024-11-09 17:51:29,925: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 17:51:29,925: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 17:51:32,124: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 17:51:32,128: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:51:32,136: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:51:32,140: INFO: common: created directory at: artifacts]
[2024-11-09 17:51:32,140: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 17:51:32,141: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 17:51:32,413: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 17:51:32,415: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 17:51:32,416: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:51:32,425: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:51:32,427: INFO: common: created directory at: artifacts]
[2024-11-09 17:51:32,427: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 17:51:37,659: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 17:51:37,662: INFO: main: >>>>>> stage Prepare Callbacks started <<<<<<]
[2024-11-09 17:51:37,665: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:51:37,673: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:51:37,675: INFO: common: created directory at: artifacts]
[2024-11-09 17:51:37,675: INFO: common: created directory at: artifacts/prepare_callbacks]
[2024-11-09 17:51:37,676: INFO: main: >>>>>> stage Prepare Callbacks completed <<<<<<

x==========x]
[2024-11-09 17:51:37,676: INFO: main: >>>>>> stage Model Training started <<<<<<]
[2024-11-09 17:51:37,680: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:51:37,690: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:51:37,692: INFO: common: created directory at: artifacts]
[2024-11-09 17:51:37,692: ERROR: main: ModelTrainingConfig.__init__() missing 5 required positional arguments: 'root_dir', 'model_path', 'epochs', 'batch_size', and 'callbacks_dir']
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 49, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_04_model_training.py", line 16, in main
    model_training_config = config_manager.get_model_training_config()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Projects\Steganalysis\src\Steganalysis\config\configuration.py", line 74, in get_model_training_config
    model_training_config = ModelTrainingConfig(
                            ^^^^^^^^^^^^^^^^^^^^
TypeError: ModelTrainingConfig.__init__() missing 5 required positional arguments: 'root_dir', 'model_path', 'epochs', 'batch_size', and 'callbacks_dir'
[2024-11-09 17:54:08,192: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 17:54:08,192: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 17:54:10,404: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 17:54:10,409: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:54:10,414: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:54:10,417: INFO: common: created directory at: artifacts]
[2024-11-09 17:54:10,417: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 17:54:10,419: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 17:54:10,696: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 17:54:10,696: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 17:54:10,700: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:54:10,706: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:54:10,711: INFO: common: created directory at: artifacts]
[2024-11-09 17:54:10,711: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 17:54:15,794: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 17:54:15,795: INFO: main: >>>>>> stage Prepare Callbacks started <<<<<<]
[2024-11-09 17:54:15,800: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:54:15,808: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:54:15,812: INFO: common: created directory at: artifacts]
[2024-11-09 17:54:15,812: INFO: common: created directory at: artifacts/prepare_callbacks]
[2024-11-09 17:54:15,814: INFO: main: >>>>>> stage Prepare Callbacks completed <<<<<<

x==========x]
[2024-11-09 17:54:15,816: INFO: main: >>>>>> stage Model Training started <<<<<<]
[2024-11-09 17:54:15,819: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:54:15,831: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:54:15,833: INFO: common: created directory at: artifacts]
[2024-11-09 17:54:15,833: ERROR: main: ModelTrainingConfig.__init__() got an unexpected keyword argument 'shuffle_data']
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 49, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_04_model_training.py", line 16, in main
    model_training_config = config_manager.get_model_training_config()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Projects\Steganalysis\src\Steganalysis\config\configuration.py", line 73, in get_model_training_config
    model_training_config = ModelTrainingConfig(
                            ^^^^^^^^^^^^^^^^^^^^
TypeError: ModelTrainingConfig.__init__() got an unexpected keyword argument 'shuffle_data'
[2024-11-09 17:56:23,087: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 17:56:23,088: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 17:56:25,454: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 17:56:25,466: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:56:25,472: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:56:25,475: INFO: common: created directory at: artifacts]
[2024-11-09 17:56:25,476: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 17:56:25,476: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 17:56:25,768: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 17:56:25,768: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 17:56:25,771: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:56:25,780: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:56:25,782: INFO: common: created directory at: artifacts]
[2024-11-09 17:56:25,783: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 17:56:31,425: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 17:56:31,425: INFO: main: >>>>>> stage Prepare Callbacks started <<<<<<]
[2024-11-09 17:56:31,429: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:56:31,434: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:56:31,436: INFO: common: created directory at: artifacts]
[2024-11-09 17:56:31,436: INFO: common: created directory at: artifacts/prepare_callbacks]
[2024-11-09 17:56:31,437: INFO: main: >>>>>> stage Prepare Callbacks completed <<<<<<

x==========x]
[2024-11-09 17:56:31,439: INFO: main: >>>>>> stage Model Training started <<<<<<]
[2024-11-09 17:56:31,439: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:56:31,448: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:56:31,448: INFO: common: created directory at: artifacts]
[2024-11-09 17:56:31,448: ERROR: main: ModelTrainingConfig.__init__() missing 2 required positional arguments: 'root_dir' and 'callbacks_dir']
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 49, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_04_model_training.py", line 16, in main
    model_training_config = config_manager.get_model_training_config()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Projects\Steganalysis\src\Steganalysis\config\configuration.py", line 73, in get_model_training_config
    model_training_config = ModelTrainingConfig(
                            ^^^^^^^^^^^^^^^^^^^^
TypeError: ModelTrainingConfig.__init__() missing 2 required positional arguments: 'root_dir' and 'callbacks_dir'
[2024-11-09 17:59:14,447: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 17:59:14,447: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 17:59:17,419: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 17:59:17,423: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:59:17,430: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:59:17,432: INFO: common: created directory at: artifacts]
[2024-11-09 17:59:17,434: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 17:59:17,435: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 17:59:17,772: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 17:59:17,772: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 17:59:17,778: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:59:17,782: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:59:17,787: INFO: common: created directory at: artifacts]
[2024-11-09 17:59:17,788: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 17:59:24,421: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 17:59:24,421: INFO: main: >>>>>> stage Prepare Callbacks started <<<<<<]
[2024-11-09 17:59:24,425: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:59:24,432: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:59:24,433: INFO: common: created directory at: artifacts]
[2024-11-09 17:59:24,433: INFO: common: created directory at: artifacts/prepare_callbacks]
[2024-11-09 17:59:24,433: INFO: main: >>>>>> stage Prepare Callbacks completed <<<<<<

x==========x]
[2024-11-09 17:59:24,433: INFO: main: >>>>>> stage Model Training started <<<<<<]
[2024-11-09 17:59:24,442: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 17:59:24,448: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 17:59:24,448: INFO: common: created directory at: artifacts]
[2024-11-09 17:59:30,022: ERROR: main: [WinError 3] The system cannot find the path specified: 'artifacts/data_ingestion/unzip_dir']
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 49, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_04_model_training.py", line 23, in main
    train_generator = data_gen.flow_from_directory(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\System_Files\Anaconda\Lib\site-packages\keras\src\legacy\preprocessing\image.py", line 1138, in flow_from_directory
    return DirectoryIterator(
           ^^^^^^^^^^^^^^^^^^
  File "E:\System_Files\Anaconda\Lib\site-packages\keras\src\legacy\preprocessing\image.py", line 453, in __init__
    for subdir in sorted(os.listdir(directory)):
                         ^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'artifacts/data_ingestion/unzip_dir'
[2024-11-09 18:02:55,296: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 18:02:55,296: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 18:02:57,607: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 18:02:57,615: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:02:57,621: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:02:57,624: INFO: common: created directory at: artifacts]
[2024-11-09 18:02:57,625: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 18:19:45,340: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 18:19:45,340: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 18:19:46,467: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 18:19:46,469: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:19:46,470: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:19:46,471: INFO: common: created directory at: artifacts]
[2024-11-09 18:19:46,471: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 18:19:46,471: INFO: data_ingestion: File already exists of size: 15.98 MB]
[2024-11-09 18:19:46,473: ERROR: main: File is not a zip file]
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 16, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_01_data_ingestion.py", line 17, in main
    data_ingestion.extract_zip_file()
  File "E:\Projects\Steganalysis\src\Steganalysis\components\data_ingestion.py", line 75, in extract_zip_file
    with zipfile.ZipFile(self.config.local_data_file, 'r') as zip_ref:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\System_Files\Anaconda\Lib\zipfile\__init__.py", line 1349, in __init__
    self._RealGetContents()
  File "E:\System_Files\Anaconda\Lib\zipfile\__init__.py", line 1416, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
[2024-11-09 18:23:18,505: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 18:23:18,505: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 18:23:19,540: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 18:23:19,540: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:23:19,544: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:23:19,544: INFO: common: created directory at: artifacts]
[2024-11-09 18:23:19,544: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 18:24:03,863: INFO: data_ingestion: artifacts\data_ingestion\data.zip download! with following info: 
Connection: close
Content-Length: 22756660
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "07b1147bb12d018cef6446014bc4d1b098f8a8db79c8f5e8866fed33abc1a055"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: E956:CF831:346E6A:41BFAE:672F567E
Accept-Ranges: bytes
Date: Sat, 09 Nov 2024 12:53:23 GMT
Via: 1.1 varnish
X-Served-By: cache-bom4735-BOM
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1731156803.012579,VS0,VE383
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 3a6b11439d70a98b309e71ecec2da8fc815f1a7e
Expires: Sat, 09 Nov 2024 12:58:23 GMT
Source-Age: 0

]
[2024-11-09 18:24:04,031: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 18:24:04,031: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 18:24:04,034: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:24:04,036: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:24:04,037: INFO: common: created directory at: artifacts]
[2024-11-09 18:24:04,037: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 18:24:06,739: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 18:24:06,741: INFO: main: >>>>>> stage Prepare Callbacks started <<<<<<]
[2024-11-09 18:24:06,743: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:24:06,746: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:24:06,746: INFO: common: created directory at: artifacts]
[2024-11-09 18:24:06,747: INFO: common: created directory at: artifacts/prepare_callbacks]
[2024-11-09 18:24:06,747: INFO: main: >>>>>> stage Prepare Callbacks completed <<<<<<

x==========x]
[2024-11-09 18:24:06,747: INFO: main: >>>>>> stage Model Training started <<<<<<]
[2024-11-09 18:24:06,749: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:24:06,753: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:24:06,754: INFO: common: created directory at: artifacts]
[2024-11-09 18:24:06,754: ERROR: main: ModelTrainingConfig.__init__() got an unexpected keyword argument 'data_path']
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 49, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_04_model_training.py", line 16, in main
    model_training_config = config_manager.get_model_training_config()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Projects\Steganalysis\src\Steganalysis\config\configuration.py", line 74, in get_model_training_config
    model_training_config = ModelTrainingConfig(
                            ^^^^^^^^^^^^^^^^^^^^
TypeError: ModelTrainingConfig.__init__() got an unexpected keyword argument 'data_path'
[2024-11-09 18:25:57,456: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 18:25:57,456: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 18:25:58,543: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 18:25:58,545: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:25:58,547: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:25:58,548: INFO: common: created directory at: artifacts]
[2024-11-09 18:25:58,548: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 18:25:58,549: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 18:25:58,689: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 18:25:58,689: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 18:25:58,691: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:25:58,693: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:25:58,694: INFO: common: created directory at: artifacts]
[2024-11-09 18:25:58,694: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 18:26:01,555: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 18:26:01,555: INFO: main: >>>>>> stage Prepare Callbacks started <<<<<<]
[2024-11-09 18:26:01,557: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:26:01,560: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:26:01,560: INFO: common: created directory at: artifacts]
[2024-11-09 18:26:01,561: INFO: common: created directory at: artifacts/prepare_callbacks]
[2024-11-09 18:26:01,561: INFO: main: >>>>>> stage Prepare Callbacks completed <<<<<<

x==========x]
[2024-11-09 18:26:01,561: INFO: main: >>>>>> stage Model Training started <<<<<<]
[2024-11-09 18:26:01,565: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:26:01,567: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:26:01,568: INFO: common: created directory at: artifacts]
[2024-11-09 18:26:03,873: ERROR: main: [WinError 3] The system cannot find the path specified: 'artifacts/data_ingestion/unzip_dir']
Traceback (most recent call last):
  File "E:\Projects\Steganalysis\main.py", line 49, in <module>
    obj.main()
  File "E:\Projects\Steganalysis\src\Steganalysis\pipeline\stage_04_model_training.py", line 31, in main
    val_generator = data_gen.flow_from_directory(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\System_Files\Anaconda\Lib\site-packages\keras\src\legacy\preprocessing\image.py", line 1138, in flow_from_directory
    return DirectoryIterator(
           ^^^^^^^^^^^^^^^^^^
  File "E:\System_Files\Anaconda\Lib\site-packages\keras\src\legacy\preprocessing\image.py", line 453, in __init__
    for subdir in sorted(os.listdir(directory)):
                         ^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'artifacts/data_ingestion/unzip_dir'
[2024-11-09 18:30:18,360: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 18:30:18,360: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 18:30:19,405: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 18:30:19,408: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:30:19,410: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:30:19,410: INFO: common: created directory at: artifacts]
[2024-11-09 18:30:19,412: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 18:30:19,412: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 18:30:19,532: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 18:30:19,533: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 18:30:19,534: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:30:19,538: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:30:19,538: INFO: common: created directory at: artifacts]
[2024-11-09 18:30:19,539: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 18:30:22,107: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 18:30:22,107: INFO: main: >>>>>> stage Prepare Callbacks started <<<<<<]
[2024-11-09 18:30:22,107: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:30:22,111: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:30:22,111: INFO: common: created directory at: artifacts]
[2024-11-09 18:30:22,113: INFO: common: created directory at: artifacts/prepare_callbacks]
[2024-11-09 18:30:22,113: INFO: main: >>>>>> stage Prepare Callbacks completed <<<<<<

x==========x]
[2024-11-09 18:30:22,114: INFO: main: >>>>>> stage Model Training started <<<<<<]
[2024-11-09 18:30:22,116: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:30:22,119: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:30:22,120: INFO: common: created directory at: artifacts]
[2024-11-09 18:44:03,869: INFO: utils: Note: NumExpr detected 16 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.]
[2024-11-09 18:44:03,871: INFO: utils: NumExpr defaulting to 8 threads.]
[2024-11-09 18:44:07,966: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-11-09 18:44:07,981: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:44:07,981: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:44:07,993: INFO: common: created directory at: artifacts]
[2024-11-09 18:44:07,993: INFO: common: created directory at: artifacts\data_ingestion]
[2024-11-09 18:44:07,995: INFO: data_ingestion: File already exists of size: 21.7 MB]
[2024-11-09 18:44:08,463: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-11-09 18:44:08,463: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-11-09 18:44:08,463: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:44:08,483: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:44:08,484: INFO: common: created directory at: artifacts]
[2024-11-09 18:44:08,486: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-11-09 18:44:14,778: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-11-09 18:44:14,778: INFO: main: >>>>>> stage Prepare Callbacks started <<<<<<]
[2024-11-09 18:44:14,778: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:44:14,798: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:44:14,801: INFO: common: created directory at: artifacts]
[2024-11-09 18:44:14,801: INFO: common: created directory at: artifacts/prepare_callbacks]
[2024-11-09 18:44:14,802: INFO: main: >>>>>> stage Prepare Callbacks completed <<<<<<

x==========x]
[2024-11-09 18:44:14,802: INFO: main: >>>>>> stage Model Training started <<<<<<]
[2024-11-09 18:44:14,804: INFO: common: yaml file: E:\Projects\Steganalysis\config\config.yaml loaded successfully]
[2024-11-09 18:44:14,811: INFO: common: yaml file: E:\Projects\Steganalysis\params.yaml loaded successfully]
[2024-11-09 18:44:14,811: INFO: common: created directory at: artifacts]
